modules/http-api/src/main/java/org/mule/runtime/http/api/client/auth/HttpAuthentication.java

```diff
@@ -91,6 +91,18 @@ static HttpNtlmAuthenticationBuilder ntlm(String username, String password) {
    */
   boolean isPreemptive();
 
+  /**
+   * Returns true if any of the provided credentials for authentications is an expression, meaning it could vary between different
+   * evaluations.
+   *
+   * @return whether the authentication credentials may vary.
+   *
+   * @since 4.3
+   */
+  default boolean forceConnectionClose() {
```

Comment:
Why is this required at the general auth level instead of just NTLM?

Reply:
Maybe it should go in the NTLMAuth. The http-connector implementation of this method doesn't go directly into the NtlmAuthentication object, but in one called UsernamePasswordAuthentication. But I think it also works if I move to NTLM. Will try.

Reply:
Just moved it.